recipes_utils.js

```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
 
-        $('.recipes__products--div').click(event => {
-            addToRecipe(event)
+    splitProductsName.forEach(productName => {
+        productName = productName.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
+        const product = $('<div>').addClass('recipes__products--div').appendTo(recipesProducts).text(productName)
 
-            $('.recipes__products--select').click(event => {
+        product.click(event => {
+            if (!$(event.target).hasClass('recipes__products--div')){
                 removeFromRecipe(event)
-            })
+                return
+            }
+            addToRecipe(event)
         })
+
     })
 
     button.click(event => {
         const recipesName = $(event.target).siblings().val()
-        const arrayOfProducts = $('.recipes__products--select').toArray()
 
-        const array = arrayOfProducts.map(object =>{
-            return object.innerHTML
-        })
+        if (recipesName !== '' && $('.recipes__products').children().hasClass('recipes__products--select')) {
+            const arrayOfProducts = $('.recipes__products--select').toArray()
+
+            const array = arrayOfProducts.map(object => {
+                return object.innerHTML
+            })
+
+            const recipe = {name: recipesName, products: array}
+
+            recipesArray = {
+                ...recipesArray,
+                recipe
+            }
+
+            $('.recipes__header').toggle('display')
+            renderRecipes(recipe.name, recipe.products)
 
-        const recipe = { name: recipesName, products: array}
+            allRecipe = allRecipe.concat(recipesArray)
 
-        recipesArray = {
-            ...recipesArray,
-            recipe
+            localStorage.setItem('recipes', JSON.stringify(allRecipe))
         }
     })
 
     return recipes
+}
+
+const renderRecipes = (name, array) => {
+    const recipesBox = $('.recipe-Box')
+    const readyRecipe = $('<div>').addClass('ready-recipe').appendTo(recipesBox)
+    const trashIcon = $('<button>').appendTo(readyRecipe).addClass('ready-recipe__trash')
+    $('<div>').addClass('ready-recipe__title').appendTo(readyRecipe).text(`name: `).css('font-weight' , 'bold')
+    $('<div>').addClass('ready-recipe__name').appendTo(readyRecipe).text(name)
+    $('<div>').addClass('ready-recipe__product').appendTo(readyRecipe).text('products: ').css('font-weight' , 'bold')
+    array.forEach( product => {
+        $('<div>').addClass('ready-recipe__product').appendTo(readyRecipe).text(product)
+    })
+
+    trashIcon.click(event => {
+        $(event.target).parents('.ready-recipe').remove()
+        const name = $(event.target).siblings('.ready-recipe__name').text()
+        allRecipe.forEach((object, index )=> {
+            if (object.recipe.name === name){
+            allRecipe.splice(index, 1)
+            }
+        })
+        localStorage.setItem('recipes', JSON.stringify(allRecipe))
```

Comment:
you can create new file and create a function to wrap localStorage and then you don't need to remember about `JSON.stringify`